File: packages/react-native-carplay/ios/RNCarPlayNavigationAlertWrapper.m
//
//  RNCarPlayNavigationAlertWrapper.m
//  RNCarPlay
//
//  Created by Manuel Auer on 24.10.24.
//
#import "RNCarPlayNavigationAlertWrapper.h"

@implementation RNCarPlayNavigationAlertWrapper

- (instancetype)initWithAlert:(CPNavigationAlert *)alert userInfo:(NSDictionary *)userInfo {
    self = [super init];
    if (self) {
        _navigationAlert = alert;
        _userInfo = userInfo;
    }
    return self;
}

@end

File: packages/react-native-carplay/ios/RNCarPlayNavigationAlertWrapper.swift
//
//  RNCarPlayNavigationAlertWrapper.swift
//  RNCarPlay
//
//  Created by Manuel Auer on 24.10.24.
//

import CarPlay

@objc(RNCarPlayNavigationAlertWrapper)
public class RNCarPlayNavigationAlertWrapper: NSObject {
    @objc public weak var navigationAlert: CPNavigationAlert?
    @objc public var userInfo: [String: Any]

    @objc public init(alert: CPNavigationAlert, userInfo: [String: Any]) {
        self.navigationAlert = alert
        self.userInfo = userInfo
    }
}
